sdk/src/main/java/com/amplitude/experiment/DefaultExperimentClient.kt

```diff
@@ -278,9 +348,11 @@ internal class DefaultExperimentClient internal constructor(
                 timeoutMillis = config.fetchTimeoutMillis
             )
         )
-        this.flags.clear()
-        this.flags.putAll(flags.get())
-        this.flags.store()
+        synchronized(this.flags) {
+            this.flags.clear()
+            this.flags.putAll(flags.get())
```

Comment:
wait for the future to complete outside the synchronized block, before clearing.